msbuild/Xamarin.MacDev.Tasks/MsBuildTasks/Copy.cs

```diff
@@ -1,12 +1,13 @@
-﻿using Xamarin.Messaging.Build.Client;
+﻿using System;
+using Xamarin.Messaging.Build.Client;
 
 namespace Microsoft.Build.Tasks
 {
 	public class Copy : CopyBase
 	{
 		public override bool Execute ()
 		{
-			if (string.IsNullOrEmpty (SessionId))
+			if (Environment.OSVersion.Platform != PlatformID.Win32NT || string.IsNullOrEmpty (SessionId))
```

Comment:
I liked the `ShouldExecuteRemotely` method, but this looks like code duplication in _a lot_ of places.

Maybe add an extension method on `Task`:

```csharp
public static bool ShouldExecuteRemotely (this Task task, string sessionId)
{
    return Environment.OSVersion.Platform != PlatformID.Win32NT || string.IsNullOrEmpty (sessionId);
}
```

and then this could become:

```suggestion
			if (ShouldExecuteRemotely (SessionId))
```

which looks much better.

Reply:
Done :)